src/tests/fs_tests.rs

```diff
@@ -805,44 +805,255 @@ pub mod fs_tests {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let path = "/nonexistent";
+        // Expect an error for non-existent path
+        assert_eq!(cage.unlink_syscall(path), -(Errno::ENOENT as i32));
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_unlink_root_directory() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let path = "/";
+        // Expect an error for unlinking root directory
+        assert_eq!(cage.unlink_syscall(path), -(Errno::EISDIR as i32));
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_unlink_directory() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        let path = "/testdir";
+        // Create the directory
+        assert_eq!(cage.mkdir_syscall(path, S_IRWXA), 0);
+
+        // Expect an error for unlinking a directory
+        assert_eq!(cage.unlink_syscall(path), -(Errno::EISDIR as i32));
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_fs_unlink_success() {
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+        let path = "/testfile";
+        // Create a file
+        let _fd = cage.open_syscall(path, O_CREAT | O_EXCL | O_WRONLY, S_IRWXA);
+        let mut statdata = StatData::default();
+        assert_eq!(cage.stat_syscall(path, &mut statdata), 0);
+        // Linkcount for the file should be 1 originally
+        assert_eq!(statdata.st_nlink, 1);
+        // Perform the unlinking of the file
+        assert_eq!(cage.unlink_syscall(path), 0);
+        // Since the file is unlinked and is not referenced by any other files
+        // Its link count should reduce to 0 and hence it should get deleted
+        // from the file system.
+        // The function below should return an error denoting no such file.
+        assert_eq!(
+            cage.stat_syscall(path, &mut statdata),
+            -(Errno::ENOENT as i32)
+        );
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
```

Comment:
Can you test that if you unlink an open file and then close it, that the space is freed?

Reply:
Sure, will update the test to check this scenario.

Reply:
Updated the test and added a new test to check for one more scenario.